core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -67,10 +68,15 @@ private void disableConnectivityReceiver() {
 
     private void setReceiverState(int receiverState) {
         if (USE_COMPONENT_ENABLED_SETTING) {
-            getPackageManager().setComponentEnabledSetting(new ComponentName(this, ConnectivityReceiver.class), receiverState, PackageManager.DONT_KILL_APP);
+            getPackageManager().setComponentEnabledSetting(connectivityReceiverComponent(), receiverState, PackageManager.DONT_KILL_APP);
         }
     }
 
+    @VisibleForTesting
+    protected ComponentName connectivityReceiverComponent() {
+        return new ComponentName(this, ConnectivityReceiver.class);
```

Comment:
I don't think so @blundell - ComponentName implementation in the android unit testing jar will be just empty so it won't matter which ctor we use because it won't assign anything.